src/expr/src/table_function/mod.rs

```diff
@@ -118,14 +115,40 @@ pub type BoxedTableFunction = Box<dyn TableFunction>;
 pub fn build_from_prost(prost: &PbTableFunction, chunk_size: usize) -> Result<BoxedTableFunction> {
     use risingwave_pb::expr::table_function::Type::*;
 
-    match prost.get_function_type().unwrap() {
-        Generate => new_generate_series::<true>(prost, chunk_size),
-        Unnest => new_unnest(prost, chunk_size),
-        RegexpMatches => new_regexp_matches(prost, chunk_size),
-        Range => new_generate_series::<false>(prost, chunk_size),
-        Udtf => new_user_defined(prost, chunk_size),
-        Unspecified => unreachable!(),
+    if prost.get_function_type().unwrap() == Udtf {
+        return new_user_defined(prost, chunk_size);
     }
+
+    build(
+        prost.get_function_type().unwrap(),
+        prost.get_return_type()?.into(),
+        chunk_size,
+        prost.args.iter().map(expr_build_from_prost).try_collect()?,
+    )
+}
+
+/// Build a table function.
+pub fn build(
+    func: PbType,
+    return_type: DataType,
+    chunk_size: usize,
+    children: Vec<BoxedExpression>,
+) -> Result<BoxedTableFunction> {
+    let args = children
+        .iter()
+        .map(|t| t.return_type().into())
+        .collect::<Vec<DataTypeName>>();
+    let desc = crate::sig::table_function::FUNC_SIG_MAP
+        .get(func, &args)
+        .ok_or_else(|| {
+            ExprError::UnsupportedFunction(format!(
+                "{:?}({}) -> {:?}",
```

Comment:
```suggestion
                "{:?}({}) -> setof {:?}",
```
?

Reply:
Thanks. It was copied from scalar function. 😅